src/run.ts

```diff
@@ -104,6 +78,44 @@ async function downloadHelm(version: string): Promise<string> {
     return helmpath;
 }
 
+async function getLatestHelmVersionFor(type) {
+    const token = core.getInput('token', { 'required': true });
+    try {
+        const { repository } = await graphql(
+            `{
+      repository(name:"helm", owner:"helm") {
+        releases(last: 100)  {
+            nodes {
+              tagName
+            }
+        }
+      }
+    }`,
+            {
+                headers: {
+                    authorization: `token ${token}`
+                }
+            }
+        );
+
+        const releases = repository.releases.nodes.reverse();
+        let latestValidRelease = releases.find(release => isValidVersion(release.tagName, type));
```

Comment:
We expect latest release to be the latest version for the type?
This seems to be a behavior change compared to the older logic which allowed all releases to compete to be the latest:
```
responseArray.forEach(response => {
   if (response && response.tag_name) {
       let currentHelmVerison = semver.clean(response.tag_name.toString());
       if (currentHelmVerison) {
           if (currentHelmVerison.toString().indexOf('rc') == -1 && semver.gt(currentHelmVerison, latestHelmVersion)) {
               //If current helm version is not a pre release and is greater than latest helm version
               latestHelmVersion = currentHelmVerison;
           }
       }
   }
});
```

Seems like a fair assumption to me. But I think this also presses the need for feature flagging to be on the safe side. Thoughts?

Reply:
Good point. But earlier we were comparing the versions so that we could get the last 3.* release of Helm. It's possible that they released a 2.* version after their latest Helm 3 release, and we didn't want to pick that up.
We can still assume that among all Helm 3.* releases, the last one would be the latest, and same goes for Helm 2.*.
Anyway, to be on the safe side I'll add a feature flag probably.